### PR TITLE
Added parsing for relative URLs

### PR DESCRIPTION
**Pull request by longchiwen**
See feature request #36 .

### CODE REVIEW

**Comment by kzap** [comment]
test this, it works

File: src/Purl/Url.php [hunk 1]
```diff
@@ -309,9 +309,14 @@ public function getNetloc()
...
     {
         $this->initialize();
-        return self::httpBuildUrl(array_map(function($value) {
+        
+        $parts = array_map(function($value) {
             return (string) $value;
-        }, $this->data));
+        }, $this->data);
+        
+        if(!$this->isAbsolute()) return self::httpBuildUrlRelative($parts);
```
Comment (jwage) [inline src/Purl/Url.php]: Format this like:

    if(!$this->isAbsolute()) {
        return self::httpBuildUrlRelative($parts);
    }

Comment (jwage) [inline src/Purl/Url.php]: Also rename the method `httpBuildUrlRelative` to `httpBuildRelativeUrl`

File: tests/Purl/Test/UrlTest.php [hunk 2]
```diff
@@ -307,6 +307,28 @@ public function testFromCurrentServerVariables() {
         $url = Url::fromCurrent();
         $this->assertEquals('http://user:passwd123@jwage.com/', (string) $url);
     }
+    
+    public function testRelativeUrl() 
+    {
+        // test all resource parts
+        $url = new Url("/path1/path2?x=1&y=2#frag");
+        $this->assertFalse($url->isAbsolute());        
+        $this->assertEquals("/path1/path2?x=1&y=2#frag",(string)$url);
```
Comment (jwage) [inline tests/Purl/Test/UrlTest.php]: Add a space after the comma and use single quotes for strings:

    $this->assertEquals('/path1/path2?x=1&y=2#frag', (string)$url);

**Comment by jwage** [comment]
:+1:

**Pull request merged**